idb/postgres/internal/util/util.go

```diff
@@ -16,10 +16,10 @@ import (
 
 // TxWithRetry is a helper function that retries the function `f` in case the database
 // transaction in it fails due to a serialization error. `f` is provided
-// a transaction created using `opts`. `f` takes ownership of the
-// transaction and must either call sql.Tx.Rollback() or sql.Tx.Commit(). In the second
-// case, `f` must return an error which contains the error returned by sql.Tx.Commit().
-// The easiest way is to just return the result of sql.Tx.Commit().
+// a transaction created using `opts`. `f` must either return an error or
+// call sql.Tx.Commit(). In the second case, `f` must return an error which contains the
+// error returned by sql.Tx.Commit(). The easiest way is to just return the result of
+// sql.Tx.Commit().
```

Comment:
I think it's fine to handle either case and describe what happens.
```suggestion
// a transaction created using `opts`. `f` should either return an error or
// call sql.Tx.Commit(). In the second case, `f` should return an error that contains the
// error returned by sql.Tx.Commit(). The easiest way is to just return the result of
// sql.Tx.Commit(). If sql.Tx.Commit() is not called, results are rolled back.
```

Reply:
done